framework/src/media/Encoder.cpp

```diff
@@ -26,18 +26,53 @@
 
 namespace media {
 
-Encoder::Encoder(audio_type_t audio_type, unsigned short channels, unsigned int sampleRate)
+Encoder::Encoder(audio_type_t audioType, unsigned short channels, unsigned int sampleRate)
 #ifdef CONFIG_AUDIO_CODEC
-	: inputBuf(nullptr), outputBuf(nullptr)
+	:
+	mAudioType(audioType),
+	mChannels(channels),
+	mSampleRate(sampleRate),
+	inputBuf(nullptr),
+	outputBuf(nullptr)
 #endif
 {
 #ifdef CONFIG_AUDIO_CODEC
-	switch (audio_type) {
+	memset(&mEncoder, 0, sizeof(audio_encoder_t));
+#endif
+}
+
+std::shared_ptr<Encoder> Encoder::create(audio_type_t audioType, unsigned short channels, unsigned int sampleRate)
+{
+#ifdef CONFIG_AUDIO_CODEC
+	medvdbg("(%d,%d,%d)\n", audioType, channels, sampleRate);
+
+	if (audioType == AUDIO_TYPE_UNKNOWN) {
+		meddbg("%s[line : %d] Fail : audio type is unknown\n", __func__, __LINE__);
+		return nullptr;
+	}
+
+	auto instance = std::make_shared<Encoder>(audioType, channels, sampleRate);
+	if (instance && instance->init()) {
+		return instance;
+	} else {
+		meddbg("%s[line : %d] Fail : init is failed\n", __func__, __LINE__);
+		meddbg("audioType : %d, channels : %d, sampleRate : %d\n", audioType, channels, sampleRate);
+		return nullptr;
+	}
+
+	meddbg("Error! (%d,%d,%d), failed!\n", audioType, channels, sampleRate);
```

Comment:
It does not call here. because of above if or else.
It is a dead code.

Reply:
Fixed it